app/code/Magento/InventorySales/Test/Integration/SalesQuoteItem/AddSalesQuoteItemOnDefaultStockTest.php

```diff
@@ -0,0 +1,204 @@
+        self::assertEquals(0, $quoteItemCount);
+
+        //cleanup
+        $this->appendReservation($productSku, 1.5);
+    }
+
+    /**
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryCatalog/Test/_files/source_items_on_default_source.php
+     */
+    public function testAddInStockProductToQuote()
+    {
+        $productSku = 'SKU-1';
+        $productQty = 4;
+        $expectedQtyInCart = 4;
+        // set reservation before (reserve -1.5 units, last 4)
+        $this->appendReservation($productSku, -1.5);
+        $product = $this->getProductBySku($productSku);
+        $quote = $this->getQuote();
+
+        $quote->addProduct($product, $productQty);
+
+        /** @var CartItemInterface $quoteItem */
+         $quoteItem = current($quote->getAllItems());
+         self::assertEquals($expectedQtyInCart, $quoteItem->getQty());
+
+        //cleanup
+        $this->appendReservation($productSku, 1.5);
+    }
+
+    /**
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryCatalog/Test/_files/source_items_on_default_source.php
+     */
+    public function testAddProductToQuoteMultipleTimes()
+    {
+        $productSku = 'SKU-1';
+        $productQty1 = 0.5;
+        $productQty2 = 3.5;
+        $productQty3 = 3.2;
+        $expectedQtyInCart1 = 0.5;
+        $expectedQtyInCart2 = 4;
+        $expectedQtyInCart3 = 4;
+        // set reservation before (reserve -1.5 units, last 4)
+        $this->appendReservation($productSku, -1.5);
+        $product = $this->getProductBySku($productSku);
+        $quote = $this->getQuote();
+
+        //last 3,5 in source
+        $quote->addProduct($product, $productQty1);
+        /** @var CartItemInterface $quoteItem */
+        $quoteItem = current($quote->getAllItems());
+        self::assertEquals($expectedQtyInCart1, $quoteItem->getQty());
+
+        //last 0 in source
+        $quote->addProduct($product, $productQty2);
+        /** @var CartItemInterface $quoteItem */
+        $quoteItem = current($quote->getAllItems());
+        self::assertEquals($expectedQtyInCart2, $quoteItem->getQty());
+
+        self::expectException(LocalizedException::class);
+        $quote->addProduct($product, $productQty3);
+        /** @var CartItemInterface $quoteItem */
+        $quoteItem = current($quote->getAllItems());
+        self::assertEquals($expectedQtyInCart3, $quoteItem->getQty());
+
+        //cleanup
+        $this->appendReservation($productSku, 1.5);
+    }
+
+    /**
+     * @param string $sku
+     * @return ProductInterface
+     * @throws NoSuchEntityException
+     */
+    private function getProductBySku(string $sku): ProductInterface
+    {
+        $product = $this->productRepository->get($sku);
+        $product->setIsSalable(true);
```

Comment:
You don't need to set is_sallable flag forcible
If product has wrong status it is a bug

Reply:
Done. Thanks.